qiskit_machine_learning/neural_networks/circuit_qnn.py

```diff
@@ -291,6 +313,12 @@ def _probability_gradients(
         # initialize empty gradients
         input_grad = None  # by default we don't have data gradients
         if self._sparse:
+            if not _HAS_SPARSE:
+                raise MissingOptionalLibraryError(
+                    libname="sparse",
+                    name="DOK",
+                    pip_install="pip install 'qiskit-machine-learning[sparse]'",
+                )
```

Comment:
Since there are two identical `if` statements in the script, might be better to have a utility method for this check? Or a global utility method for such checks.

Reply:
Since you need to import sparse at the file either at top level or at the moment it is used, I don't see you gaining much. You would have an import inside some utility to test sparse existence and throw the exception with a custom name, in this case 'DOK'. Then at that place in the code or at top level you still need to import SparseArray/DOK/COO and make sure to let the ImportError pass so that you can use it and then call the utility method. IN the current case, you only have the top import with the _HAS_SPARSE and the tests raising the exception.